lib/chef/provider/package/windows.rb

```diff
@@ -56,24 +50,62 @@ def package_provider
           @package_provider ||= begin
             case installer_type
             when :msi
+              require 'chef/provider/package/windows/msi'
               Chef::Provider::Package::Windows::MSI.new(resource_for_provider)
             else
```

Comment:
today they would have to be treated as a custom type. my guess is that `MSU`s have the same "product" semantics as MSIs. I'll test and if so we could juse the  MSI class for MSUs.